flink-table/flink-table-common/src/test/java/org/apache/flink/table/types/extraction/TypeInferenceExtractorTest.java

```diff
@@ -861,4 +907,32 @@ public Number eval(Number n) {
 			return n;
 		}
 	}
+
+	@DataTypeHint("ROW<i INT>")
+	private static class DataTypeHintOnTableFunction extends TableFunction<Row> {
```

Comment:
Just a nit, but this could be `DataTypeHintOnTableFunctionClass` and the next one could be ``DataTypeHintOnTableFunctionMethod`.